core/src/smartcontracts/isi/query.rs

```diff
@@ -12,16 +15,16 @@ pub trait Lazy {
     type Lazy<'a>;
 }
 
-/// Lazily evaluated equivalent of [`Value`]
-pub enum LazyValue<'a> {
-    /// Concrete computed [`Value`]
-    Value(Value),
-    /// Iterator over a set of [`Value`]s
-    Iter(Box<dyn Iterator<Item = Value> + 'a>),
+/// Lazily evaluated equivalent of [`Query::Output`]
+pub enum LazyQueryOutput<'a> {
```

Comment:
there is no need. Where would you like to use it?

Reply:
Then it could be `pub(crate)`

Reply:
privacy policy is mostly irrelevant for `iroha_core` because it's not a part of the public API. Also, changing this to `pub(crate)` would be a big change all across the core crate because it's used in pub functions